clang/lib/Sema/SemaDecl.cpp

```diff
@@ -12486,6 +12486,14 @@ void Sema::CheckMain(FunctionDecl *FD, const DeclSpec &DS) {
                                 : FixItHint());
       FD->setInvalidDecl(true);
     }
+
+    // In C++ [basic.start.main]p3, it is said a program attaching main to a
+    // named module is ill-formed.
```

Comment:
nit: generally we we'll cite it directly:

```suggestion
    // [basic.start.main]p3:
    //
    //       ....
```